k2/csrc/fsa_algo.cu

```diff
@@ -203,13 +203,17 @@ bool Intersect(FsaOrVec &a_fsas, int32_t properties_a, FsaOrVec &b_fsas,
     int32_t this_num_arcs = creator.GetArcOffsetFor(i + 1) - arc_offset;
     if (arc_map_a) {
       int32_t arc_offset_a = a_fsas_row_splits12_data[i * stride_a];
-      for (int32_t i = 0; i < this_num_arcs; i++)
-        this_arc_map_a[i] += arc_offset_a;
+      for (int32_t i = 0; i < this_num_arcs; i++) {
+        this_arc_map_a[i] = this_arc_map_a[i] == -1 ? this_arc_map_a[i] :
```

Comment:
I would suggest
<img width="729" alt="Screen Shot 2021-07-06 at 3 50 38 PM" src="https://user-images.githubusercontent.com/5284924/124562727-f2234300-de71-11eb-9c7c-b68004f8494b.png">
which is clearer, I think. You can keep it if you like it.